src/lib/util.ts

```diff
@@ -90,16 +90,22 @@ export async function blobToImg(blob: Blob): Promise<HTMLImageElement> {
     const img = new Image();
     img.decoding = 'async';
     img.src = url;
+    const loaded = new Promise((resolve, reject) => {
+      img.onload = () => resolve();
+      img.onerror = () => reject(Error('Image loading error'));
+    });
 
     if (img.decode) {
-      // Nice off-thread way supported in at least Safari.
-      await img.decode();
+      // Nice off-thread way supported in Safari/Chrome.
+      try {
+        await img.decode();
+      } catch (err) {
+        // Safari throws on decode if the source is SVG. Fallback to onload.
+        // https://bugs.webkit.org/show_bug.cgi?id=188347
+        await loaded;
+      }
     } else {
-      // Main thread decoding :(
-      await new Promise((resolve, reject) => {
-        img.onload = () => resolve();
-        img.onerror = () => reject(Error('Image loading error'));
-      });
+      await loaded;
```

Comment:
Looking at the control flow here, I _think_ we can simplify and remove the entire `if` statement.

```js
try {
	await img.decode();
} 
catch(e) {}
finally {
	await loaded;
}

```

Reply:
I think it can be even simpler than that. I think I was having a bad code day.